packages/shared/clickhouse/migrations/unclustered/0033_tool_columns.up.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE observations ADD COLUMN tool_definitions Map(String, String) DEFAULT map();
```

Comment:
Migrations themselves look alright.

A few considerations:
1. What happened to the arrays idea?
2. I would very strongly suggest splitting migration commits from the rest of this PR.
3. You would have to add these columns to the `events` table as well and coordinate with @Steffen911 on double-writes.